staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2466,12 +2468,21 @@ type InstancetypeMatcher struct {
 	//
 	// +optional
 	RevisionName string `json:"revisionName,omitempty"`
+
+	// InferFromVolume lists the name of a volume that should be used to infer or discover the instancetype
+	// to be used through known annotations on the underlying resource. Once applied to the InstancetypeMatcher
+	// this field is removed.
+	//
+	// +optional
+	InferFromVolume string `json:"inferFromVolume,omitempty"`
```

Comment:
I'm unsure about the behavior here where the `InferFromVolume` is mutated and replaced by the `Name` and `Kind` fields.

For example, what should we expect to happen if someone replaces their root volume?  They declared that they wanted preferences/instancetypes from a volume, but we anchor to a name/kind from then on out.

Maybe this is okay. I'd just like to see it discussed a bit.

Reply:
The current behaviour came from a previous suggestion to be as deterministic as possible, so any mutations of the underlying annotated resources wouldn't impact existing `VirtualMachines`.

That said there's nothing stopping the user from mutating the `{Instancetype,Preference}Matcher` to infer from the new volume again if they want to.